### PR TITLE
[eos-noon] cmake refactoring + cpack

### PR DESCRIPTION
**Pull request by asiniscalchi**
I did the same changes in the master branch but, as eos-noon will be the new master, I re-propose them

### CODE REVIEW

**Comment by asiniscalchi** [comment]
image

File: CMakeModules/installer.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+set(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR}/install)
+
+SET(CPACK_PACKAGE_DIRECTORY "${CMAKE_INSTALL_PREFIX}")
+set(CPACK_PACKAGE_NAME "eos")
+set(CPACK_PACKAGE_VENDOR "EOSIO")
+set(CPACK_PACKAGE_VERSION_MAJOR "${VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${VERSION_PATCH}")
+set(CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
+set(CPACK_PACKAGE_DESCRIPTION "A client for the Eos network")
```
Comment (wanderingbort) [inline CMakeModules/installer.cmake]: We've had an official update on the name and this should be "EOS.IO"

Comment (asiniscalchi) [reply CMakeModules/installer.cmake]: Package vendor or name or in description?

File: CMakeModules/doxygen.cmake [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  set(DOXY_DOC_DEST_DIR "${CMAKE_BINARY_DIR}/docs" CACHE PATH "Path to the doxygen output")
+  set(DOXY_DOC_INPUT_ROOT_DIR "contracts" CACHE PATH "Path to the doxygen input")
+  if(DOXYGEN_DOT_FOUND)
+    set(DOXY_HAVE_DOT "YES" CACHE STRING "Doxygen to use dot for diagrams.")
+  else(DOXYGEN_DOT_FOUND)
+    set(DOXY_HAVE_DOT "NO" CACHE STRING "Doxygen to use dot for diagrams.")
+  endif(DOXYGEN_DOT_FOUND)
+  configure_file("eos.doxygen.in" "${CMAKE_BINARY_DIR}/eos.doxygen")
+  if(BUILD_DOXYGEN)
+     # !! Core developers who don't want to see the doxygen output on every build
```
Comment (jgiszczak) [inline CMakeModules/doxygen.cmake]: Please incorporate existing changes in the doxygen-related code in this branch.

File: CMakeModules/installer.cmake (hunk 1 above)
Comment (wanderingbort) [reply CMakeModules/installer.cmake]: name and description.  Vendor can be "block.one".  I think @jgiszczak  is determining the proper package contact address.

Comment (jgiszczak) [reply CMakeModules/installer.cmake]: The package contact address should be "support@block.one".

Comment (asiniscalchi) [reply CMakeModules/installer.cmake]: done

File: CMakeModules/doxygen.cmake (hunk 2 above)
Comment (asiniscalchi) [reply CMakeModules/doxygen.cmake]: done

**Comment by asiniscalchi** [comment]
image

File: CMakeModules/installer.cmake [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
-set(CPACK_PACKAGE_CONTACT "someone@eos.org")
+set(CPACK_PACKAGE_CONTACT "support@block.one")
 set(CPACK_OUTPUT_FILE_PREFIX ${CMAKE_BINARY_DIR}/packages)
 set(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR}/install)
 
 SET(CPACK_PACKAGE_DIRECTORY "${CMAKE_INSTALL_PREFIX}")
-set(CPACK_PACKAGE_NAME "eos")
-set(CPACK_PACKAGE_VENDOR "EOSIO")
+set(CPACK_PACKAGE_NAME "EOS.IO")
+set(CPACK_PACKAGE_VENDOR "black.one")
```
Comment (wanderingbort) [inline CMakeModules/installer.cmake]: block.one not, black.one

**Pull request merged**